### PR TITLE
feat: support pkgfile frontend.

### PR DESCRIPTION
**Pull request by frezbo**
This supports `Pkgfile` projects.

Targets can be set as below in `.kres.yaml`:

```yaml
---
kind: pkgfile.Build
spec:
  targets:
    - toolchain
  reproducibleTargetName: toolchain
  additionalTargets:
    nonfree:
      - nonfree-kmod-nvidia-pkg
```

### CODE REVIEW

File: internal/output/dockerignore/dockerignore.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (frezbo) [inline internal/output/dockerignore/dockerignore.go:1]: this was split from `dockerfile` to `dockerignore`

File: internal/project/auto/golang.go [hunk 2]
```diff
@@ -22,6 +23,10 @@ import (
 
 // DetectGolang checks if project at rootPath is Go-based project.
 func (builder *builder) DetectGolang() (bool, error) {
+	if builder.meta.ContainerImageFrontend != config.ContainerImageFrontendDockerfile {
```
Comment (frezbo) [inline internal/project/auto/golang.go:26]: we can short circuit early, also helps adding conditional checks in so many other places

File: internal/project/auto/markdown.go [hunk 3]
```diff
@@ -22,23 +23,25 @@ func (builder *builder) DetectMarkdown() (bool, error) {
 		}
 	}
 
-	{
-		list, err := listFilesWithSuffix(builder.rootPath, ".md")
-		if err != nil {
-			return false, err
-		}
+	list, err := listFilesWithSuffix(builder.rootPath, ".md")
```
Comment (frezbo) [inline internal/project/auto/markdown.go:26]: this was just cleanup, the previous code was wrapped in `{}`

Comment (smira) [reply internal/project/auto/markdown.go:26]: it was wrapped for readability, probably unsuccessful attempt to make it more readable :stuck_out_tongue_winking_eye:

File: internal/project/common/all.go [hunk 4]
```diff
@@ -28,6 +29,10 @@ func NewAll(meta *meta.Options) *All {
 
 // CompileMakefile implements makefile.Compiler.
 func (all *All) CompileMakefile(output *makefile.Output) error {
+	if all.meta.ContainerImageFrontend != config.ContainerImageFrontendDockerfile {
```
Comment (smira) [inline internal/project/common/all.go:32]: this _should_ still work same way for `Pkgfile` builds, given that the tree is built correctly

Comment (frezbo) [reply internal/project/common/all.go:32]: it was adding `all: pkgfile build`

not sure where it came from (`pkgfile` was the dag name)

Comment (smira) [reply internal/project/common/all.go:32]: scratch that

**Review by smira — APPROVED** [review]

**Pull request merged**